.buildkite/commands/install-node-dependencies.sh

```diff
@@ -0,0 +1,49 @@
+#!/bin/bash -eu
+
+PLATFORM=$(uname -s)
+ARCHITECTURE=$(uname -m)
+NODE_VERSION=$(node --version)
+PACKAGE_HASH=$(hash_file package-lock.json)
+
+CACHEKEY="$BUILDKITE_PIPELINE_SLUG-npm-$PLATFORM-$ARCHITECTURE-node-$NODE_VERSION-$PACKAGE_HASH"
+
+LOCAL_NPM_CACHE=./vendor/npm
+mkdir -p $LOCAL_NPM_CACHE
+echo "--- :npm: Set npm to use $LOCAL_NPM_CACHE for cache"
+npm set cache $LOCAL_NPM_CACHE
+echo "npm cache set to $(npm get cache)"
+
+echo "--- :npm: Restore npm cache if present"
+restore_cache "$CACHEKEY"
+
+echo "--- :npm: Install Node dependencies"
+
+MAX_SOCKETS=15 # Default value from npm
+
+# To avoid constant ECONNRESET errors a limit is set for Linux,
+# as this is not happening with the Mac jobs.
+# This issue is being tracked here:
+# https://github.com/npm/cli/issues/4652
+if [ "$PLATFORM" = "Linux" ]; then
+  MAX_SOCKETS=1
+fi
+
+npm ci \
+  --unsafe-perm \
+  --prefer-offline \
+  --no-audit \
+  --no-progress \
+  --maxsockets "$MAX_SOCKETS" \
+  "$@"
+
+echo "--- :npm: Save cache if necessary"
+# Notice that we don't cache the local node_modules.
+# Those get regenerated by npm ci as per Node reccomendations.
+# What we are caching is the root npm folder, which stores pacakge downloads so they are available if the package.json resolution demands them.
```

Comment:
This is very interesting information. I wonder why we didn't go that route in https://github.com/Automattic/a8c-ci-toolkit-buildkite-plugin/blob/trunk/bin/install_npm_packages; might be worth updating that helper to make other apps using `npm` benefit from this?